plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,44 +186,96 @@ export class WinstonInstrumentation extends InstrumentationBase {
     return (original: Winston2LogMethod) => {
       const instrumentation = this;
       return function patchedLog(
-        this: unknown,
+        this: never,
         ...args: Parameters<typeof original>
       ) {
-        const span = trace.getSpan(context.active());
-
-        if (!span) {
-          return original.apply(this, args);
-        }
-
-        const spanContext = span.spanContext();
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
+              record = injectRecord(spanContext, record);
+            }
+            instrumentation._callHook(span, record);
 
-        if (!isSpanContextValid(spanContext)) {
-          return original.apply(this, args);
-        }
+            // Inject in metadata argument
+            let isDataInjected = false;
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                args[i] = Object.assign(args[i], record);
+                isDataInjected = true;
+                break;
+              }
+            }
+            if (!isDataInjected) {
+              const insertAt =
+                typeof args[args.length - 1] === 'function'
+                  ? args.length - 1
+                  : args.length;
 
-        for (let i = args.length - 1; i >= 0; i--) {
-          if (typeof args[i] === 'object') {
-            const record = args[i];
-            injectRecord(spanContext, record);
-            instrumentation._callHook(span, record);
-            return original.apply(this, args);
+              args.splice(insertAt, 0, record);
```

Comment:
we are manipulating the `args` here, and then use the updated value below for the log bridge. Is it ok?

Reply:
Well the span information would be duplicated in this case I guess if Logger is also adding correlation information into the LogRecords